### PR TITLE
feat(create): use prompts package and refactor flow

### PR DESCRIPTION
**Pull request by daKmoR**
This implements the flow as outlined in the issue https://github.com/open-wc/open-wc/issues/376

- replaced `qoa` with `prompts`
- added minimal virtual file system
- added tests (some not all 🙈 )
- added full control via command line

it works like this
create-flow

if that sounds good I can update the readme + all the other packages

PS: with the virtual file system we can later add an output tree and questions if people want to override existing files 👍 

closes https://github.com/open-wc/open-wc/issues/376, https://github.com/open-wc/open-wc/issues/370

### CODE REVIEW

**Comment by daKmoR** [comment]
we can also do a pre-release if anyone would like to have a simple way of testing it out -> just ask 🤗

**Comment by LarsDenBakker** [comment]
Looks good to me!

File: packages/create/src/generators/app/header.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export default `
+        _.,,,,,,,,,._
+     .d''           \`\`b.
+   .p'      Open       \`q.        Open Web Components Create
+ .d'    Web Components  \`b.
+ .d'                     \`b.      It let's you scaffold or upgrade web components projects.
```
Comment (thepassle) [inline packages/create/src/generators/app/header.js]:
```suggestion
 .d'                     \`b.      It lets you scaffold or upgrade your web component projects.
```

File: packages/create/src/generators/app/header.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export default `
+        _.,,,,,,,,,._
+     .d''           \`\`b.
+   .p'      Open       \`q.        Open Web Components Create
```
Comment (thepassle) [inline packages/create/src/generators/app/header.js]:
```suggestion
   .p'      Open       \`q.        Open Web Component Recommendations
```

File: packages/create/src/generators/app/header.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+ .d'    Web Components  \`b.
+ .d'                     \`b.      It let's you scaffold or upgrade web components projects.
+ ::   .................   ::       You can call it with...
```
Comment (thepassle) [inline packages/create/src/generators/app/header.js]: This sentence seems odd to me?

File: packages/create/src/generators/app/header.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+ \`p.                     .q'
+  \`p.    open-wc.org    .q'       Details for the generator can be found at
```
Comment (thepassle) [inline packages/create/src/generators/app/header.js]:
```suggestion
  \`p.    open-wc.org    .q'       Details for the generators can be found at
```

File: packages/create/src/generators/app/index.js [hunk 5]
```diff
@@ -1,102 +1,191 @@
...
-            'Enterprise App (if you feel lost use the Starter App first)',
-            'Lit Element Web Component',
-            'Mono Repo for web components',
+          type: (prev, all) => (all.type === 'scaffold' ? 'select' : null),
+          name: 'scaffoldType',
+          message: 'What would you like to scaffold?',
+          choices: [
+            { title: 'Web Component', value: 'wc' },
+            { title: 'Application', value: 'app' },
+            { title: 'Starter App (pre configured)', value: 'starter-app' },
```
Comment (thepassle) [inline packages/create/src/generators/app/index.js]:
```suggestion
            { title: 'Starter App', value: 'starter-app' },
```

**Comment by thepassle** [comment]
Awesome! Looks so nice! 😍

File: packages/create/src/generators/app/header.js (hunk 3 above)
Comment (daKmoR) [reply packages/create/src/generators/app/header.js]: yeah this whole header is just a WIP 🙈

Comment (thepassle) [reply packages/create/src/generators/app/header.js]: I think we can just omit it

Comment (daKmoR) [reply packages/create/src/generators/app/header.js]: adjusted the header - pls take another look 🤗

File: packages/create/src/generators/app/header.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import chalk from 'chalk';
+
+export default `
+        _.,,,,,,,,,._
+     .d''           \`\`b.       ${chalk.underline('Open Web Components Recommendations')}
+   .p'      Open       \`q.
+ .d'    Web Components  \`b.    Start or upgrade your web component project with
+ .d'                     \`b.   ease. All the nice setting at your fingertips.
```
Comment (thepassle) [inline packages/create/src/generators/app/header.js]: Maybe "All our recommendations at your fingertips?"

File: packages/create/src/generators/app/header.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+ ::   .................   ::
+ \`p.                     .q'   See more details at https://open-wc.org/create/
+  \`p.    open-wc.org    .q'
+   \`b.     @openWc     .d'
+     \`q..            ..,'      Note: you can exit any time with Strg+C or Esc
```
Comment (thepassle) [inline packages/create/src/generators/app/header.js]:
```suggestion
     \`q..            ..,'      Note: you can exit any time with Ctrl+C or Esc
```

(is strg german? :p)

Comment (daKmoR) [reply packages/create/src/generators/app/header.js]: jup 🙈

**Pull request merged**